### PR TITLE
[msbuild/dotnet] Add support for NativeReference items. Fixes #11061.

### LINKED ISSUE #11061: @(NativeReference) does nothing in .NET 6
### Steps to Reproduce

1. Create a new .NET 6 iOS app (I used HelloiOS from the net6 samples)
2. Add a `<NativeReference Include="..." Kind="Framework" />` to the project
3. Compile
4. Observe an error `ld: framework not found libSkiaSharp`

I originally noticed this with Mac Catalyst and then went to try iOS.

### Expected Behavior

App compiles and runs. I have a classic Xamarin.iOS app and that ran fine.

> I did notice that I had an issue when I used a relative path to the framework in that, but that is another issue and solvable with the `$(MSBuildThisFileDirectory)`.

### Actual Behavior

Build error and I also noticed that it did not get the framework in the Frameworks folder in the .app

### Environment

```
.NET 6 SDK: `6.0.100-preview.2.21155.3`
iOS SDK: `14.4.100-ci.main.1192`
```

### Build Logs

[msbuild.binlog.zip](https://github.com/xamarin/xamarin-macios/files/6240154/msbuild.binlog.zip)

### Example Project (If Possible)

[Archive.zip](https://github.com/xamarin/xamarin-macios/files/6240152/Archive.zip)

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Pass any native references to the LinkNativeCode task so that they're linked
  with the main executable.
* Copy frameworks and dynamic libraries to the app bundle (also add support
  for MSBuild metadata - the CopyToAppBundle property - to avoid copying a
  framework / dynamic library to the app bundle).
* Refactor MTouchTaskBase a bit to make parts of it reusable from the
  LinkNativeCode task.
* Add tests.

Fixes https://github.com/xamarin/xamarin-macios/issues/11061.

### CODE REVIEW

File: tests/dotnet/NativeFileReferencesApp/Makefile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	rm -Rf */bin */obj
+
+all-ios: prepare
+	$(DOTNET6) build iOS/*.csproj /bl
+
+all-mac: prepare
+	$(DOTNET6) build macOS/*.csproj /bl
+
+run-mac:
+	/Users/rolf/work/maccore/whatever/xamarin-macios/tests/dotnet/NativeReferencesApp/macOS/bin/Debug/net6.0-macos/osx-x64/NativeReferences.app/Contents/MacOS/NativeReferences
```
Comment (spouliot) [inline tests/dotnet/NativeFileReferencesApp/Makefile:20]: should be relative to `$(TOP)`
your's too far away, well beyond `MAX_PATH` ;-)

**Review by emaf — APPROVED** [review]

File: tests/dotnet/NativeFrameworkReferencesApp/Makefile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	rm -Rf */bin */obj
+
+all-ios: prepare
+	$(DOTNET6) build iOS/*.csproj /bl
+
+all-mac: prepare
+	$(DOTNET6) build macOS/*.csproj /bl
+
+run-mac:
+	/Users/rolf/work/maccore/whatever/xamarin-macios/tests/dotnet/NativeReferencesApp/macOS/bin/Debug/net6.0-macos/osx-x64/NativeReferences.app/Contents/MacOS/NativeReferences
```
Comment (spouliot) [inline tests/dotnet/NativeFrameworkReferencesApp/Makefile:20]: same

**Review by spouliot — CHANGES_REQUESTED** [review]
This seems to cover xcframework but there's no associated tests ported (or added) to dotnet

File: tests/dotnet/NativeFileReferencesApp/Makefile (hunk 1 above)
Comment (rolfbjarne) [reply tests/dotnet/NativeFileReferencesApp/Makefile]: > your's too far away, well beyond `MAX_PATH` ;-)

🤦

**Comment by rolfbjarne** [comment]
@spouliot 

> This seems to cover xcframework but there's no associated tests ported (or added) to dotnet

I've added a test case for xcframeworks as well (and as expected, it wasn't working at first, I had to fix something).

**Review by spouliot — APPROVED** [review]
❤️ it

**Comment by rolfbjarne** [comment]
Test failure is unrelated (https://github.com/xamarin/maccore/issues/2414).

**Pull request merged**